sentry/src/main/java/io/sentry/clientreport/AtomicClientReportStorage.java

```diff
@@ -0,0 +1,76 @@
+package io.sentry.clientreport;
+
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.transport.DataCategory;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class AtomicClientReportStorage implements ClientReportStorage {
+
+  private final @NotNull Map<ClientReportKey, AtomicLong> lostEventCounts;
+
+  public AtomicClientReportStorage() {
+    Map<ClientReportKey, AtomicLong> map = new ConcurrentHashMap<>();
+
+    for (DiscardReason discardReason : DiscardReason.values()) {
+      for (DataCategory category : DataCategory.values()) {
```

Comment:
```suggestion
    for (final DiscardReason discardReason : DiscardReason.values()) {
      for (final DataCategory category : DataCategory.values()) {
```

I like having `final` always when its a immutable var, so when you read the code, you know this is not changing its value.